Public/Src/Cache/ContentStore/App/KustoUploader.cs

```diff
@@ -0,0 +1,167 @@
+    ///       (2) posting multiple files from concurrent threads is ok.
+    ///
+    ///     To complete ingestion and wait for all pending ingestion tasks to finish, call
+    ///     <see cref="CompleteAndWaitForPendingIngestionsToFinish"/>.  After this method has been
+    ///     called, <see cref="PostFileForIngestion(string, Guid)"/> does not accept any new requests.
+    ///
+    ///     The <see cref="Dispose"/> method of this class implicitly waits for pending ingestions
+    ///     to finish (by calling <see cref="CompleteAndWaitForPendingIngestionsToFinish"/>.
+    /// </summary>
+    public sealed class KustoUploader : IDisposable
+    {
+        private readonly ILog _log;
+        private readonly bool _deleteFilesOnSuccess;
+        private readonly bool _checkForIngestionErrors;
+        private readonly IKustoQueuedIngestClient _client;
+        private readonly KustoQueuedIngestionProperties _ingestionProperties;
+        private readonly ActionBlock<FileDescription> _block;
+
+        /// <summary>
+        ///     Constructor.  Initializes this object and does nothing else.
+        /// </summary>
+        /// <param name="connectionString">Kusto connection string.</param>
+        /// <param name="database">Database into which to ingest.</param>
+        /// <param name="table">Table into which to ingest.</param>
+        /// <param name="deleteFilesOnSuccess">Whether to delete files upon successful ingestion.</param>
+        /// <param name="checkForIngestionErrors">Whether to check for ingestion errors before disposing this object.</param>
+        /// <param name="log">Optional log to which to write some debug information.</param>
+        public KustoUploader
+            (
+            string connectionString,
+            string database,
+            string table,
+            bool deleteFilesOnSuccess,
+            bool checkForIngestionErrors,
+            ILog log = null
+            )
+        {
+            _log = log;
+            _deleteFilesOnSuccess = deleteFilesOnSuccess;
+            _checkForIngestionErrors = checkForIngestionErrors;
+            _client = KustoIngestFactory.CreateQueuedIngestClient(connectionString);
+            _ingestionProperties = new KustoQueuedIngestionProperties(database, table)
+            {
+                ReportLevel = IngestionReportLevel.FailuresOnly,
+                ReportMethod = IngestionReportMethod.Queue
+            };
+            _block = new ActionBlock<FileDescription>
+                (
+                    (fileDesc) =>
+                    {
+                        var start = DateTime.UtcNow;
+                        var result = _client.IngestFromSingleFile(fileDesc, _deleteFilesOnSuccess, _ingestionProperties);
+                        var duration = DateTime.UtcNow.Subtract(start);
+
+                        Info("Ingesting file '{0}' took {1}", fileDesc.FilePath, duration);

```

Comment:
>{1} [](start = 56, length = 3)

Could we standardize this and use `duration.TotalMilliseconds`ms ? Or something similar?

Reply:
will do